transactions/src/main/java/org/jboss/as/txn/service/internal/tsr/JCAOrderedLastSynchronizationList.java

```diff
@@ -164,29 +157,7 @@ public void afterCompletion(int status) {
             } catch (Exception e) {
                 // Trap these exceptions so the rest of the synchronizations get the chance to complete
                 // https://github.com/jbosstm/narayana/blob/5.0.4.Final/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/SynchronizationImple.java#L102
-                TransactionLogger.ROOT_LOGGER.jcaSyncAfterCompletionFailed(jcaSync, tx, e);
-            }
-        }
-
-        if (jcaOrderedLastSynchronizations.remove(tx) == null) {
```

Comment:
Where does this bit end up going?

Reply:
It was added for JTS mode by the look of it. Sometimes in JTS we can't get the ID of the transaction (for equality test) in afterCompletion - it depends if it has been required earlier as it is cached when retrieved but not until then.

Reply:
Instead of having a map of transaction (ID) to list, I put the list directly on the transaction.

Reply:
Ah, right I see. The main thing is to not rely on the equality during or after afterCompletion as the underlying data may not have been retrieved. Assuming that is not the case here I am approving.